Shared/Bytes.swift

```diff
@@ -11,8 +11,13 @@ public typealias GUID = String
  */
 open class Bytes {
     open class func generateRandomBytes(_ len: UInt) -> Data {
-        let bytes = [UInt32](repeating: 0, count: Int(len)).map { _ in arc4random() }
-        return Data(bytes: bytes, count: Int(len))
+        let len = Int(len)
+        var data = Data(count: len)
+        data.withUnsafeMutableBytes { (p: UnsafeMutablePointer<UInt8>) in
+            let result = SecRandomCopyBytes(kSecRandomDefault, len, p)
+            assert(result == 0, "Random byte generation failed.")
```

Comment:
That would be fine: `precondition` and `fatalError` are mostly equivalent.

https://github.com/apple/swift/blob/adc54c8a4d13fbebfeb68244bac401ef2528d6d0/stdlib/public/core/Assert.swift#L170
https://github.com/apple/swift/blob/adc54c8a4d13fbebfeb68244bac401ef2528d6d0/stdlib/public/core/Assert.swift#L191

In a release build, `precondition` simply int-traps without printing. I don't think we care about that distinction.